tools/pipelines/test-dds-stress.yml

```
@@ -22,38 +22,87 @@ resources:
 variables:
 - name: testWorkspace
   value: $(Pipeline.Workspace)/test
+- name: absolutePathToTelemetryGenerator
+  value: $(Build.SourcesDirectory)/tools/telemetry-generator
+  readonly: true
+
+parameters:
+- name: packages
+  type: object
+  default:
+  - name: "@fluidframework/sequence"
+    affectedPaths:
+    - packages/dds/sequence
+    - packages/dds/merge-tree
+    testFileTarName: sequence
+    testCommand: test:stress
+  - name: "@fluid-experimental/tree"
+    affectedPaths:
+    - experimental/dds/tree
+    testFileTarName: experimental-tree
+    testCommand: test:stress
+  - name: "@fluidframework/tree"
+    affectedPaths:
+    - packages/dds/tree
+    testFileTarName: tree
+    testCommand: test:stress
+  - name: "@fluidframework/map"
+    affectedPaths:
+    - packages/dds/map
+    - packages/dds/merge-tree
+    testFileTarName: map
+    testCommand: test:stress
+  - name: "@fluidframework/matrix"
+    affectedPaths:
+    - packages/dds/matrix
+    - packages/dds/merge-tree
+    testFileTarName: matrix
+    testCommand: test:stress
+
 
 stages:
   - template: templates/include-conditionally-run-stress-tests.yml
     parameters:
       artifactBuildId: $(resources.pipeline.client.runID)
-      packages:
-      - name: "@fluidframework/sequence"
-        affectedPaths:
-        - packages/dds/sequence
-        - packages/dds/merge-tree
-        testFileTarName: sequence
-        testCommand: test:stress
-      - name: "@fluid-experimental/tree"
-        affectedPaths:
-        - experimental/dds/tree
-        testFileTarName: experimental-tree
-        testCommand: test:stress
-      - name: "@fluidframework/tree"
-        affectedPaths:
-        - packages/dds/tree
-        testFileTarName: tree
-        testCommand: test:stress
-      - name: "@fluidframework/map"
-        affectedPaths:
-        - packages/dds/map
-        - packages/dds/merge-tree
-        testFileTarName: map
-        testCommand: test:stress
-      - name: "@fluidframework/matrix"
-        affectedPaths:
-        - packages/dds/matrix
-        - packages/dds/merge-tree
-        testFileTarName: matrix
-        testCommand: test:stress
+      packages: ${{ parameters.packages }}
       testWorkspace: ${{ variables.testWorkspace }}
+
+  # Capture telemetry about pipeline stages
+  - stage: FollowupStage
```

Nit, for consistency with the other pipelines
```suggestion
  - stage: upload_run_telemetry
```